monitor.py

```diff
@@ -1306,16 +1333,26 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
 
             elif event_type == 'navroute':
                 # Added in ED 3.7 - multi-hop route details in NavRoute.json
-                with open(join(self.currentdir, 'NavRoute.json'), 'rb') as rf:  # type: ignore
-                    try:
-                        entry = json.load(rf)
+                if (nv_json := self._parse_navroute_file()) is not None:
+                    entry_time = mktime(strptime(nv_json['timestamp'], '%Y-%m-%dT%H:%M:%SZ'))
+                    c_time = time()
```

Comment:
This check should be versus the `timestamp` in the Journal event that signalled an updated file.

Local time *can* be substantially off from the server time that's used in the Journal files (and I assume also in the additional files).  So this could easily have a false positive when trying to detect "that's an old file".